internal/jimm/model.go

```diff
@@ -245,6 +245,7 @@ func (b *modelBuilder) WithCloudRegion(region string) *modelBuilder {
 				continue
 			}
 			region = r.Name
+			break
```

Comment:
i've added this break, because before it seemed to me it was wrong to continue the loop if you've already found a region.
Let me know if we were relying on this behaviour for some reason